File: Src/java/engine/src/test/java/org/opencds/cqf/cql/engine/execution/Issue223.java
package org.opencds.cqf.cql.engine.execution;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

import java.util.List;
import org.junit.jupiter.api.Test;

class Issue223 extends CqlTestBase {

    @Test
    void interval() {
        var results = engine.evaluate(toElmIdentifier("Issue223"));
        var value = results.forExpression("Access Flattened List of List Items").value();
        List<?> list = (List<?>) value;
        assertThat(list.size(), is(1));
        assertThat(list.get(0), is(true));

        value = results.forExpression("Access Flattened List of List Items in a Single Query")
                .value();
        list = (List<?>) value;
        assertThat(list.size(), is(1));
        assertThat(list.get(0), is(true));
    }
}

File: Src/java/engine/src/jvmTest/kotlin/org/opencds/cqf/cql/engine/execution/Issue223.kt
package org.opencds.cqf.cql.engine.execution

import kotlin.test.assertEquals
import kotlin.test.assertIs
import org.junit.jupiter.api.Test
import org.opencds.cqf.cql.engine.runtime.Boolean
import org.opencds.cqf.cql.engine.runtime.List

internal class Issue223 : CqlTestBase() {
    @Test
    fun interval() {
        val results = engine.evaluate { library("Issue223") }.onlyResultOrThrow
        var value = results["Access Flattened List of List Items"]!!.value
        assertIs<List>(value)
        assertEquals(1, value.count())
        assertEquals(Boolean.TRUE, value.elementAt(0))

        value = results["Access Flattened List of List Items in a Single Query"]!!.value
        assertIs<List>(value)
        assertEquals(1, value.count())
        assertEquals(Boolean.TRUE, value.elementAt(0))
    }
}
